### ISSUE TITLE
Database constantly grows in size when using dexie-observable

### ISSUE THREAD

**Issue by opus1269** [dfahlander/Dexie.js#598]
The indexedDB is constantly growing at about 1KB per second when using dexie-observable in multiple windows even though the app is not adding data.  This size increase persists across browser sessions.

If I remove dexie-observable and all calls to db.on('changes', xxx); the db size remains stable.

Here is a video of devtools showing the growth:
https://drive.google.com/file/d/0B8EmrW2mMp9GRkxzRWs0YjNialk/view

You can see it for yourself by installing the following extension in Chrome:
https://chrome.google.com/webstore/detail/pushy-clipboard/jemdfhaheennfkehopbpkephjlednffd

Info.
Dexie version: 2.0.1
Chrome version: 62.0.3202.62 (Official Build) (64-bit)
OS: Windows 10 creators
Chrome extension with persistent background page (i.e. always loaded)

**Comment by dfahlander** [comment]
Thanks for sharing! I'll try looking into it later this week.
/David

**Comment by dfahlander** [comment]
I was unable to reproduce it with the latest version of your chrome extension. Maybe you've removed the use of dexie-observable in the latest version?

Anyway, there is an explanation to the temporary increase. Dexie-observable maintains communication between tabs/windows (called syncNodes internally) using heartbeat messages that aren't consumed until a tab is active, or fails to respond in a timely manner. These messages are cleaned up after a while (after 20-60 seconds). A slight increase may persist if closing down all tabs at once but once next time app is started, these messages will be cleared out.

**Comment by opus1269** [comment]
Yes. I switched to hooks with messaging between windows, as needed. The thing is, it was never cleaning up, it would grow unconditionally over time until Chrome was exited. Maybe the problem is that an extension's background page isn't a tab so is never "active". In my case, I also use the webRequest api, which also requires the background page be persistent so it never unloads. (Event pages are loaded on demand and unloaded after a brief period of inactivity).

Anyways, I think this is still going to be an issue for unsuspecting extension developers. I didn't notice it until I started get QuotaErrors logged from users shortly after I first implemented dexie-observable.

**Comment by dfahlander** [comment]
Ok, I see. Good that you found your way through crud hooks. I am currently in the beginning of taking the next steps for dexie-observable and dexie-syncable and there might be need of a rewrite and simplification for them. I'll take this issue into account. Thanks!
